cylc/flow/config.py

```diff
@@ -815,6 +820,25 @@ def process_start_cycle_point(self) -> None:
             # Start from the initial point.
             self.start_point = self.initial_point
 
+    def cycle_point_warning(self, label1, order, label2):
+        """Centralized logic for warning that start, stop, initial and
+        final cycle points are not sensibly ordered.
+        """
+        point1 = getattr(self, label1 + '_point')
+        point2 = getattr(self, label2 + '_point')
+        if (
+            order == 'before' and point1 < point2
+            or order == 'after' and point1 > point2
+        ):
+            msg = (
+                f"{label1} cycle point '{point1}' will have no effect as"
+                f" it is {order} the {label2} cycle point '{point2}'."
```

Comment:
I think there is conflation of 2 different cases here.

1. Valid but useless X cycle point. E.g.:
    > start cycle point '1' before the initial cycle point '2'

    This should give a warning.

2. Invalid X cycle point. E.g.:
    > stop cycle point '1' before the initial cycle point '2'

    I think this ought to raise an input error.

Reply:
A legit complaint. Happy to rework this.